### PR TITLE
V1.2.5 blend rcp channel meta with system channel meta

### PR DESCRIPTION
**Pull request by brentpicasso**
This is an effort to keep the runtime channel meta data in sync with the set of reference system channels.

If an RCP is connected to the system, the channel configuration is blended with the system channel defaults and additional custom channels are also added, for a "runtime channel meta data" configuration.

This has the effect of gauges showing the proper range and units that have been defined in RCP's channel configuration and generally keeping things in sync.

### CODE REVIEW

File: autosportlabs/racecapture/data/sampledata.py [hunk 1]
```diff
@@ -128,10 +129,57 @@ def processData(self, dataJson):
...
+        #first, reload the default set of channels
+        self.reload_system_channels()
+        
+        channels = self.channels
+        #now override defaults with current runtime values
+        for channel_name, runtime_meta in runtime_channel_meta.iteritems():
+            existing_meta = channels.get(channel_name)
+            
+            if existing_meta is not None:
+                #override existing cahnnel meta with channel meta from runtime
```
Comment (ryandoherty) [inline autosportlabs/racecapture/data/sampledata.py]: Typo!

File: autosportlabs/racecapture/views/dashboard/widgets/fontgraphicalgauge.py [hunk 2]
```diff
@@ -8,12 +8,17 @@ class FontGraphicalGauge(GraphicalGauge):
...
+        
+    def on_max(self, instance, value):
+        self._refresh_gauge()
+        
     def updateColors(self):
         self.graphView.color = self.select_alert_color()
         return super(FontGraphicalGauge, self).updateColors()
         
-    def refreshGaugeDisplay(self):
+    def _refresh_gauge(self):
```
Comment (ryandoherty) [inline autosportlabs/racecapture/views/dashboard/widgets/fontgraphicalgauge.py]: What's the reason for the leading underscore?

**Comment by ryandoherty** [comment]
So, how can I test these changes to view the new functionality or at least verify old functionality didn't break?

File: autosportlabs/racecapture/data/sampledata.py (hunk 1 above)
Comment (brentpicasso) [inline autosportlabs/racecapture/data/sampledata.py]: >_<

File: autosportlabs/racecapture/views/dashboard/widgets/fontgraphicalgauge.py (hunk 2 above)
Comment (brentpicasso) [inline autosportlabs/racecapture/views/dashboard/widgets/fontgraphicalgauge.py]: indicates a function should be treated privately

Comment (ryandoherty) [inline autosportlabs/racecapture/views/dashboard/widgets/fontgraphicalgauge.py]: :+1:

**Comment by brentpicasso** [comment]
The purpose is of this to ensure configured gauges (e.g. a round gauge with 0-100%) is automatically updated when you go in and change some aspect of the channel. For example, change min / max from 0-100 to 0-500. That kind of stuff.

In other words, channel meta data is automatically reflected in the dashboard when an RCP is connected.

**Comment by brentpicasso** [comment]
Also, this has updates to the base gauge class so it responds and updates to channel meta changes as they happen.  In RCP- if you change a config, it triggers a refresh in the metadata / channel configuration. Eg. if you add/remove a channel, or you change channel name or min/max - that kind of stuff.

**Comment by ryandoherty** [comment]
So I setup a custom analog channel with a random name, saved config, set a round gauge to show it, then changed the name and saved the config again. The gauge name did not update:

Config after updating:
http://i.imgur.com/CaSGJ86.png

Dash after updating:
http://i.imgur.com/m7z0OwQ.png

**Comment by brentpicasso** [comment]
You saved it by writing the config to RCP?


On Wed, Mar 25, 2015 at 10:35 PM, Ryan Doherty <notifications@github.com>
wrote:

> So I setup a custom analog channel with a random name, saved config, set a
> round gauge to show it, then changed the name and saved the config again.
> The gauge name did not update:
>
> Config after updating:
> http://i.imgur.com/CaSGJ86.png
>
> Dash after updating:
> http://i.imgur.com/m7z0OwQ.png
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/autosportlabs/RaceCapture_App/pull/25#issuecomment-86349999>
> .
>



-- 
Brent Picasso
*Autosport Labs <http://www.autosportlabs.com>*
*Technology for Race and Street*

**Comment by ryandoherty** [comment]
Yup!

**Comment by brentpicasso** [comment]
Re-read what you wrote. My changes doesn't cover that use case. I think by
design. You changed the channel name on RCP side; the dashboard gauge is
mapped to watch for a certain channel, regardless if it's present or not in
the data stream.

On Wed, Mar 25, 2015 at 11:19 PM, Brent Picasso <brent@autosportlabs.com>
wrote:

> You saved it by writing the config to RCP?
>
>
> On Wed, Mar 25, 2015 at 10:35 PM, Ryan Doherty <notifications@github.com>
> wrote:
>
>> So I setup a custom analog channel with a random name, saved config, set
>> a round gauge to show it, then changed the name and saved the config again.
>> The gauge name did not update:
>>
>> Config after updating:
>> http://i.imgur.com/CaSGJ86.png
>>
>> Dash after updating:
>> http://i.imgur.com/m7z0OwQ.png
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/autosportlabs/RaceCapture_App/pull/25#issuecomment-86349999>
>> .
>>
>
>
>
> --
> Brent Picasso
> *Autosport Labs <http://www.autosportlabs.com>*
> *Technology for Race and Street*
>



-- 
Brent Picasso
*Autosport Labs <http://www.autosportlabs.com>*
*Technology for Race and Street*

**Comment by brentpicasso** [comment]
try changing something like Units or min/max


On Wed, Mar 25, 2015 at 11:20 PM, Ryan Doherty <notifications@github.com>
wrote:

> Yup!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/autosportlabs/RaceCapture_App/pull/25#issuecomment-86358172>
> .
>



-- 
Brent Picasso
*Autosport Labs <http://www.autosportlabs.com>*
*Technology for Race and Street*

**Comment by ryandoherty** [comment]
Yeah, I tested changing units and min/max, that worked :)

**Comment by brentpicasso** [comment]
K, works as coded!

On Wed, Mar 25, 2015 at 11:23 PM, Ryan Doherty <notifications@github.com>
wrote:

> Yeah, I tested changing units and min/max, that worked :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/autosportlabs/RaceCapture_App/pull/25#issuecomment-86358407>
> .
>



-- 
Brent Picasso
*Autosport Labs <http://www.autosportlabs.com>*
*Technology for Race and Street*

**Comment by ryandoherty** [comment]
:shipit:

**Pull request merged**